test/e2e/config/azure-dev.yaml

```diff
@@ -3,11 +3,11 @@ managementClusterName: capz-e2e
 images:
   - name: ${MANAGER_IMAGE}
     loadBehavior: mustLoad
-  - name: registry.k8s.io/cluster-api/cluster-api-controller:v1.5.2
+  - name: gcr.io/k8s-staging-cluster-api/cluster-api-controller:nightly_main_20230929
```

Comment:
We had to pin the test version of CAPI to the nightly build as MachinePool workers support is not in the latest release of CAPI (1.5)

Reply:
Seems fine to me, maybe add a comment explaining the reason for that? And the condition for when we can go back to the normal build.

Reply:
Yes we decided in CAPZ office hours that it would be okay to merge this change since we are only using the newer CAPI binaries for testing.